### ISSUE TITLE
Consider removing dependence on time

### ISSUE THREAD

**Issue by mbarger99** [aphysci/gravity#148]
Currently Gravity depends on time in several places.  A couple examples:
 * Incoming messages are rejected if they are older than messages previously seen from the same publisher.  
 * If multiple ServiceDirectory's advertise themselves on the same domain, the newest one is chosen by any listening GravityNode's.

This has worked well for us as long as the system clocks on the machines in use don't jump backwards.  In some cases though, startup issues with a system clock may cause it to jump around some, and ultimately settle on a time that's earlier than it was previously set to.  If this happens all subsequent messages will be rejected by subscribers as they think messages from publishers on that machine are old.  

I think it's worth looking at our use of time here to see if the benefits outweigh this issue.

**Comment by chrismurf** [comment]
I agree.  We should use the monotonic clock for any internal timers / comparisons, and consider reworking/removing any logic that relies on external clocks.
